src/tactic/ring.lean

```diff
@@ -468,9 +473,9 @@ local postfix `?`:9001 := optional
 /-- Tactic for solving equations in the language of rings.
   This version of `ring` fails if the target is not an equality
   that is provable by the axioms of commutative (semi)rings. -/
-meta def ring1 : tactic unit :=
+meta def ring1 (red : transparency) : tactic unit :=
```

Comment:
This should also use `parse (lean.parser.tk "!")?` - `ring1` is also a frontend tactic